### PR TITLE
Added code to search tags, #PG-3288

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Added code to search tags
Fixes: #PG-3288, #754

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: vue/src/Tag/TagList.vue [hunk 1]
```diff
@@ -390,14 +404,31 @@ export default defineComponent({
       return TagsStore.tags.value;
     },
     sortedTags() {
-      const sorted = [...this.tags];
-      sorted.sort((lhs, rhs) => {
+      const searchFilter = this.tagSearch.toLowerCase();
+
+      // look through string properties of custom reports for values that have searchFilter in them
```
Comment (snake14) [inline vue/src/Tag/TagList.vue]: Not a big deal, but I'm guessing this is a copy/paste error?
```suggestion
      // look through string properties of tags for values that have searchFilter in them
```

**Review by snake14 — APPROVED** [review]
Looks good! I had one minor comment.

Aside from that, I noticed that triggers in the Except list weren't included in the search. It's probably not a big deal, though.
image

image

File: vue/src/Tag/TagList.vue (hunk 1 above)
Comment (AltamashShaikh) [reply vue/src/Tag/TagList.vue]: Thanks done :+1:

**Comment by AltamashShaikh** [comment]
> Looks good! I had one minor comment.
> 
> Aside from that, I noticed that triggers in the Except list weren't included in the search. It's probably not a big deal, though. image
> 
> image

It can have multiple values since its an array, so haven't include that, but I can include it in the trigger and variable search ticket.

**Pull request merged**